pennylane/transforms/core/__init__.py

```diff
@@ -15,4 +15,12 @@
 
 from .transform import transform
 from .transform_dispatcher import TransformDispatcher, TransformContainer, TransformError
-from .transform_program import TransformProgram
+from .compile_pipeline import CompilePipeline
+
+
+def __getattr__(key):
+    if key == "TransformProgram":
+        return CompilePipeline
+    raise AttributeError(
+        f"module 'pennylane.transforms.core' has no attribute '{key}'"
+    )  # pragma: no cover
```

Comment:
We can also just do:

```suggestion
TransformProgram = CompilePipeline
```

and switch to the `__getattr__` when we need to raise a deprecation warning on access.